tools/fuzzer/expr_gen.cc

```diff
@@ -118,7 +119,7 @@ std::optional<Expr> ExprGenerator::gen_integer_constant(
 
 std::optional<Expr> ExprGenerator::gen_double_constant(
     const ExprConstraints& constraints) {
-  if (constraints.must_be_lvalue()) {
+  if (constraints.must_be_lvalue() || constraints.must_be_valid_pointer()) {
```

Comment:
Floating points literals cannot be implicitly converted to pointers, so I guess, technically, this can never be true here? Maybe assert on it instead?

Reply:
That makes sense, this should already be regulated by `type_constraints`. Added assert in the `if` below.